docs/src/idiomatic/assignments.md

```diff
@@ -0,0 +1,129 @@
+# Idiom 1: Update state variables with assignments
+For each TLA+ variable `x` defined in a specification, Apalache requires that every possible  transition defines an assignment to `x'`.
+
+## Assignments and Assignment Candidates
+An _assignment candidate_ for `x` is an expression that has one of the following forms:
+- `x' = e`, 
+- `x' \in e`,
+- `UNCHANGED x`, or
+- `x' := e`
+
+While a transition may contain multiple assignment candidates for the same variable, not all of them are chosen as _assignments_ by Apalache. The sections below describe how the assignments are selected.
+
+### Minimality
+Assignments aren't spurious; each variable must have at least one assignment per transition operator, but no more than necessary to satisfy all of the additional constraints below.
+
+If all possible transitions fail to assign one or more variables, an exception of the following shape is thrown:
```

Comment:
We really have to be clear about what transitions are. I would say that we start with actions, which is a standard terminology of TLA+. Then we can say that sometimes actions have different options. So we split actions into smaller ones, to have one assignment per refined action. This is what we call symbolic transitions. Perhaps, we should get rid of the transition terminology at all. Let's invent a name, like 'minimal actions' or something.

Reply:
In either case, since this is not something that is defined in Lamport's book, it seems like there needs to be some sort of chapter dedicated to explaining transitions/minimal actions, but I think this chapter should be its own PR.